sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -27,21 +27,45 @@ import org.apache.spark.sql.catalyst.util.{ArrayData, DateTimeUtils, MapData}
 import org.apache.spark.sql.types._
 
 private[sql] class JacksonGenerator(
-    schema: StructType,
+    childType: DataType,
```

Comment:
This parameter looks weird to me. We don't want to tightly link `JacksonGenerator` to what this PR did.

Reply:
ok.  I recover it and use `rowSchema` to take what `MapType` needs.